python/lsst/ts/tcpip/client.py

```diff
@@ -0,0 +1,262 @@
+from __future__ import annotations
+
+# This file is part of ts_tcpip.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["Client"]
+
+import asyncio
+import logging
+import types
+import typing
+
+from . import utils
+from .constants import DEFAULT_MONITOR_CONNECTION_INTERVAL
+
+
+class Client:
+    """A TCP/IP socket client.
+
+    A thin wrapper around `asyncio.open_connection`. Like that function,
+    it can only be used to connect once. Construct a new instance each
+    time you wish to make a new connection.
+
+    Parameters
+    ----------
+    host : `str` | `None`
+        IP address.
+    port : `int` | `None`
+        IP port.
+    log : `logging.Logger`
+        Logger.
+    connect_callback : callable or `None`, optional
```

Comment:
It should be more flexible if the `connect_callback` can be synchronous or asynchronous.